test/src/module/command/backupCommonTest.c

```diff
@@ -322,5 +329,53 @@ testRun(void)
         TEST_RESULT_STR_Z(backupTypeStr(backupTypeIncr), "incr", "backup type str incr");
     }
 
+    // *****************************************************************************************************************************
+    if (testBegin("backupLinkLatest()"))
+    {
+        String *backupStanzaPath = strNew("repo/backup/db");
+        String *backupLatest = strNew("20181119-152800F");
+
+        StringList *argList = strLstNew();
+        strLstAddZ(argList, "--stanza=db");
+        strLstAdd(argList, strNewFmt("--repo1-path=%s/repo", testPath()));
+        harnessCfgLoad(cfgCmdArchiveGet, argList);
+
+        // -------------------------------------------------------------------------------------------------------------------------
+        TEST_TITLE("error on latest link creation");
+
+        TEST_ERROR_FMT(
+            backupLinkLatest(backupLatest), FileOpenError,
+            "unable to create symlink '%s/%s/" BACKUP_LINK_LATEST "' to '%s': [2] No such file or directory", testPath(),
+            strPtr(backupStanzaPath), strPtr(backupLatest));
+
+        // -------------------------------------------------------------------------------------------------------------------------
+        TEST_TITLE("path sync not enabled (code coverage)");
+
+        // Create a backup directory by creating a file
+        storagePutP(
+            storageNewWriteP(storageTest, strNewFmt("%s/%s/test" , strPtr(backupStanzaPath), strPtr(backupLatest))),
+            BUFSTRDEF("tmp"));
+
+        // Disable storage features for code coverage
+        ((Storage *)storageRepoWrite())->interface.feature ^= 1 << storageFeaturePathSync;
+        ((Storage *)storageRepoWrite())->interface.feature ^= 1 << storageFeatureSymLink;
+
+        TEST_RESULT_VOID(backupLinkLatest(backupLatest), "attempt to create backup latest symlink");
+        TEST_RESULT_BOOL(
+            storageExistsP(storageTest, strNewFmt("%s/" BACKUP_LINK_LATEST, strPtr(backupStanzaPath))), false,
+            "link does not exist");
+
+        // Enable storage features
+        ((Storage *)storageRepoWrite())->interface.feature |= 1 << storageFeaturePathSync;
+        ((Storage *)storageRepoWrite())->interface.feature |= 1 << storageFeatureSymLink;
+
+        // -------------------------------------------------------------------------------------------------------------------------
+        TEST_TITLE("latest link creation success");
+
+        TEST_RESULT_VOID(backupLinkLatest(backupLatest), "create backup latest symlink");
+        TEST_RESULT_STR(storageInfoP(storageRepo(), STRDEF(STORAGE_REPO_BACKUP "/" BACKUP_LINK_LATEST)).linkDestination,
+            backupLatest, "latest link set");
+    }
+
```

Comment:
I'm not excited about a new test suite for code that already had complete coverage.

I've been thinking of combining the backup-common and backup test suites. Should we go ahead and do that? I think it is fair to argue that it is a "unit".

One of the reasons we've split up test suites in the past is that our test macros were so expensive that the compilers were running out of memory. We have since fixed that.

Reply:
Fixed in commit: 9dfbac06b7e6f5e26759e3be4246d19078dc5659